### PR TITLE
Enables EF DbContext pooling.

### PR DESCRIPTION
**Pull request by anpete**
cc @mikeharder, @nathana1 

Depends on https://github.com/aspnet/EntityFramework/tree/fix-6923

### CODE REVIEW

File: src/Benchmarks/Startup.cs [hunk 1]
```diff
@@ -51,23 +52,35 @@ public IServiceProvider ConfigureServices(IServiceCollection services)
...
+                            if (appSettings.Database == DatabaseServer.PostgreSql)
+                            {
+                                ob.UseNpgsql(appSettings.ConnectionString);
+                            }
+                            else
+                            {
+                                ob.UseSqlServer(appSettings.ConnectionString);
+                            }
+                        },
+                poolSize: 32);
```
Comment (mikeharder) [inline src/Benchmarks/Startup.cs]: How should applications determine the optimal pool size?  Might a larger size be better?

Comment (anpete) [inline src/Benchmarks/Startup.cs]: Good question. It would be great to get some data on different sizes during testing of this change.

**Comment by anpete** [comment]
Update: This has landed in EF dev. I need to tweak this PR due to some API changes.

**Comment by anpete** [comment]
New version up.

**Comment by mikeharder** [comment]
I benchmarked this PR on Azure F4 VMs.  Results:

Scenario | Baseline | PR | Improvement
------------ | ------------- | ------------ | -------------
Single Query (/db/ef) | 9,103 | 11,864 | 30%
Multiple Queries (/queries/ef?queries=20) | 730 | 755 | 3%
Multiple Updates (/updates/ef?queries=20) | 564 | 580 | 3%
Fortunes (/fortunes/ef) | 8,809 | 10,984 | 25%

**Pull request merged**